database/settings/2024_04_23_111934_add_change_management_in_license_add_ons.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2024, Canyon GBS LLC. All rights reserved.
+
+    Aiding App™ is licensed under the Elastic License 2.0. For more details,
+    see <https://github.com/canyongbs/aidingapp/blob/main/LICENSE.>
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Aiding App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    <https://www.canyongbs.com> or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+use App\Settings\LicenseSettings;
+use Spatie\LaravelSettings\Migrations\SettingsMigration;
+
+return new class () extends SettingsMigration {
+    public function up(): void
+    {
+        $data = app(LicenseSettings::class)->data;
+        $data = @json_decode(json_encode($data), true);
+
+        /** Add new settings property */
+        $data['addons']['change_management'] = true;
+
+        /** Deleting existing data group */
+        $this->migrator->delete('license.data');
+
+        /** Creating License data group again with new propery with existing data */
+        $this->migrator->addEncrypted(
+            'license.data',
+            $data
+        );
```

Comment:
I believe that this can use `updateEncrypted()`

Reply:
Actually I think you can just save the value
```php
$licenseSettings = app(LicenseSettings::class);
$licenseSettings->data->addons->changeManagement = true;
$licenseSettings->save();
```

Reply:
I don't think you should interact with the settings class in a migration, that kinda breaks our guidelines

Reply:
Ah, if this is intended to be data migration you think something like this?
```php
$this->migrator->updateEncrypted([
  'license.data.addons.change_management' => true,
]);
```

Reply:
> Ah, if this is intended to be data migration you think something like this?
> 
> ```
> $this->migrator->updateEncrypted([
>   'license.data.addons.change_management' => true,
> ]);
> ```

HI @mxm1070 

If we update like this then it will be give error 

Spatie\LaravelSettings\Migrations\SettingsMigrator::updateEncrypted(): Argument #1 ($property) must be of type string, array given, called in /var/www/html/database/settings/2024_04_24_082011_license_settings.php on line 8

if we follow the syntax from the doc 

```
public function up(): void
{
    $this->migrator->updateEncrypted(
        'general.site_name', 
        fn(string $siteName) => return 'Space'
    );
}
```

and do something like that 

```
 $this->migrator->updateEncrypted(
            'license.data.addons.test_ankit',
            fn () => true
        );
```
        
It will also throw the same error, hence I need to use that way.